### PR TITLE
Refactor ReferenceWidget

### PR DESCRIPTION
**Pull request by Tishasoumya-02**
#4460

### CODE REVIEW

File: packages/volto/news/5093.feature [hunk 1]
```diff
@@ -0,0 +1 @@
+Refactor ReferenceWidget from class to functional component. @Tishasoumya
```
Comment (stevepiercy) [inline packages/volto/news/5093.feature:1]:
```suggestion
Refactor `ReferenceWidget` from class-based to functional component. @Tishasoumya
```

**Review by stevepiercy — DISMISSED** [review]

**Review by stevepiercy — COMMENTED** [review]
I only reviewed news. Still needs a technical review.

File: packages/volto/src/components/manage/Widgets/ReferenceWidget.jsx [hunk 2]
```diff
@@ -216,59 +137,71 @@ class ReferenceWidget extends Component {
...
+    PropTypes.object,
+  ]),
+  onChange: PropTypes.func.isRequired,
+  search: PropTypes.arrayOf(
+    PropTypes.shape({
+      '@id': PropTypes.string,
+      '@type': PropTypes.string,
+      title: PropTypes.string,
+      description: PropTypes.string,
     }),
```
Comment (davisagli) [inline packages/volto/src/components/manage/Widgets/ReferenceWidget.jsx]:
```suggestion
```

`search` now comes from useSelector and is not passed as a prop

File: packages/volto/src/components/manage/Widgets/ReferenceWidget.jsx [hunk 3]
```diff
@@ -216,59 +137,71 @@ class ReferenceWidget extends Component {
...
   ),
-)(ReferenceWidget);
+  wrapped: PropTypes.bool,
+};
+
+ReferenceWidget.defaultProps = {
+  description: null,
+  required: false,
+  error: [],
+  search: [],
```
Comment (davisagli) [inline packages/volto/src/components/manage/Widgets/ReferenceWidget.jsx]:
```suggestion
```

**Review by davisagli — APPROVED** [review]

**Pull request merged**